docs/api/specs/desktopAgentCommunicationProtocol.md

```diff
@@ -0,0 +1,448 @@
+
+Request and response used to implement the [`getAppMetadata()`](../ref/DesktopAgent#getappmetadata) API call:
+
+- [`getAppMetadataRequest`](https://fdc3.finos.org/schemas/next/api/getAppMetadataRequest.schema.json)
+- [`getAppMetadataResponse`](https://fdc3.finos.org/schemas/next/api/getAppMetadataResponse.schema.json)
+
+#### `getCurrentChannel()`
+
+Request and response used to implement the [`getCurrentChannel()`](../ref/DesktopAgent#getcurrentchannel) API call:
+
+- [`getCurrentChannelRequest`](https://fdc3.finos.org/schemas/next/api/getCurrentChannelRequest.schema.json)
+- [`getCurrentChannelResponse`](https://fdc3.finos.org/schemas/next/api/getCurrentChannelResponse.schema.json)
+
+#### `getInfo()`
+
+Request and response used to implement the [`getInfo()`](../ref/DesktopAgent#getinfo) API call:
+
+- [`getInfoRequest`](https://fdc3.finos.org/schemas/next/api/getInfoRequest.schema.json)
+- [`getInfoResponse`](https://fdc3.finos.org/schemas/next/api/getInfoResponse.schema.json)
+
+#### `getOrCreateChannel()`
+
+Request and response used to implement the [`getOrCreateChannel()`](../ref/DesktopAgent#getorcreatechannel] API call:
+
+- [`getOrCreateChannelRequest`](https://fdc3.finos.org/schemas/next/api/getOrCreateChannelRequest.schema.json)
+- [`getOrCreateChannelResponse`](https://fdc3.finos.org/schemas/next/api/getOrCreateChannelResponse.schema.json)
+
+#### `getUserChannels()`
+
+Request and response used to implement the [`getUserChannels()`](../ref/DesktopAgent#getuserchannels) API call:
+
+- [`getUserChannelsRequest`](https://fdc3.finos.org/schemas/next/api/getUserChannelsRequest.schema.json)
+- [`getUserChannelsResponse`](https://fdc3.finos.org/schemas/next/api/getUserChannelsResponse.schema.json)
+
+#### `joinUserChannel()`
+
+Request and response used to implement the [`joinUserChannel()`](../ref/DesktopAgent#joinchannel) API call:
+
+- [`joinUserChannelRequest`](https://fdc3.finos.org/schemas/next/api/joinUserChannelRequest.schema.json)
+- [`joinUserChannelResponse`](https://fdc3.finos.org/schemas/next/api/joinUserChannelResponse.schema.json)
+
+#### `leaveCurrentChannel()`
+
+Request and response used to implement the [`leaveCurrentChannel()`](../ref/DesktopAgent#leavecurrentchannel) API call:
+
+- [`leaveCurrentChannelRequest`](https://fdc3.finos.org/schemas/next/api/leaveCurrentChannelRequest.schema.json)
+- [`leaveCurrentChannelResponse`](https://fdc3.finos.org/schemas/next/api/leaveCurrentChannelResponse.schema.json)
+
+#### `open()`
+
+Request and response used to implement the [`open()`](../ref/DesktopAgent#open) API call:
+
+- [`openRequest`](https://fdc3.finos.org/schemas/next/api/openRequest.schema.json)
+- [`openResponse`](https://fdc3.finos.org/schemas/next/api/openResponse.schema.json)
+
+Where a context object is passed (e.g. `fdc3.open(app, context)`) the `broadcastEvent` message described above in [`addContextListener`](#addcontextlistener) should be used to deliver it after the context listener has been added:
```

Comment:
should we also describe what happens if no `addContextListenerRequest` is received - I assume the context is just not sent or is an error returned to the openRequest caller?

Reply:
You are correct (`OpenError.AppTimeout` is the error that should be returned/open promise rejected with) - and I note I've missed the openResponse out. I've added some text on this, added a tip that links to docs on the timeouts and corrected the diagram:

![image](https://github.com/user-attachments/assets/d64e710e-c660-4c65-b420-82c4cd2f4735)